From c5ea4f999ec563e2df674d77b5f92daecc28a110 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 24 Jan 2024 03:49:01 +0100
Subject: [PATCH] polkit: Reset the idle_id before emitting the signal

The authenticator object is destroyed in the signal handler, so we need to reset the idle_id before emitting the signal.

https://gitlab.gnome.org/GNOME/gnome-flashback/-/issues/92
---
 daemons/polkit/flashback-authenticator.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/daemons/polkit/flashback-authenticator.c b/daemons/polkit/flashback-authenticator.c
index f384119..d5b780e 100644
--- a/daemons/polkit/flashback-authenticator.c
+++ b/daemons/polkit/flashback-authenticator.c
@@ -197,12 +197,12 @@ do_initiate (gpointer user_data)
   if (!flashback_polkit_dialog_run_until_user_is_selected (dialog))
     {
       authenticator->was_cancelled = TRUE;
+      authenticator->idle_id = 0;
 
       g_signal_emit_by_name (authenticator, "completed",
                              authenticator->gained_authorization,
                              authenticator->was_cancelled);
 
-      authenticator->idle_id = 0;
       return G_SOURCE_REMOVE;
     }
 
@@ -275,11 +275,12 @@ do_initiate (gpointer user_data)
       break;
     }
 
+  authenticator->idle_id = 0;
+
   g_signal_emit_by_name (authenticator, "completed",
                          authenticator->gained_authorization,
                          authenticator->was_cancelled);
 
-  authenticator->idle_id = 0;
   return G_SOURCE_REMOVE;
 }
 
-- 
2.44.0

